Add code to solve the Sudoku puzzle

cclauss opened the pull request:
This should close issue https://github.com/prajwalkr/SnapSudoku/issues/2

Commit:
Add code to solve the Sudoku puzzle

This should close issue https://github.com/prajwalkr/SnapSudoku/issues/2

Commit:
Update sudoku_str.py

prajwalkr commented:
Before solving, we would need to check for validity of the predicted puzzle, devise an effective method to remove all conflicts. (two equal digits in same row/col/subgrid)

Also, we can save the Sudoku solving script offline in the scripts/ folder ourselves, why programmatically ask for an internet connection and do it.

I'll merge this PR once a method for conflict-resolution is found :smile:

Commit:
Update sudoku_str.py

cclauss commented:
Added sudo.py to scripts.
Removed download code.
Todo: must add a test before trying to solve.  sudopy.random_puzzle() seems to have such a test.  I will try to figure that out.

cclauss commented:
Done.
